lab_1/scripts/update-pipeline-definition.sh

```diff
@@ -0,0 +1,36 @@
+cd ..
+
+if ! command -v jq &> /dev/null
+  then
+    'jq' does not exists
+    exit
+fi
+
+JSON_FILE=''
+BRANCH='main'
+OWNER='owner'
+
+if [[ $# -eq 1 || $# -gt 1 ]];
+  then
+    JSON_FILE=$1
+fi
+
+while ! [ $# -eq 1 ]; do
+  case $1 in
+    '--branch')
+      BRANCH=$2
+      ;;
+    '--owner')
+      OWNER=$2
+      ;;
+  esac
+  shift
+done
+
+if [ $JSON_FILE = '' ];
+  then
+    echo Provide path to json file
+    exit
+fi
+
+jq 'del(.metadata)' $JSON_FILE | jq '.pipeline.version += 1' | jq --arg branch $BRANCH '.pipeline.stages[0].actions[0].configuration.Branch |= $branch' | jq --arg owner $OWNER '.pipeline.stages[0].actions[0].configuration.Owner |= $owner' > "pipeline-$( date ).json"
```

Comment:
1. There is no validation for metadata and version properties.

"The script should validate if the necessary properties are present in the given JSON definition. If not, it should throw an error and stop execution."

2. you stick to particular stage and particular action index.

There is no action name filtering (Source action should be changed only):

"The Branch property in the Source action’s configuration is set to a value from the script’s parameter/flag --branch. "

3. No solution provided for "PollForSourceChanges" and "EnvironmentVariables"